soh/soh/Enhancements/randomizer/randomizer_entrance.c

```diff
@@ -524,3 +546,61 @@ void Entrance_OverrideSpawnScene(s32 sceneNum, s32 spawn) {
         }
     }
 }
+
+u8 Entrance_GetIsSceneDiscovered(u8 sceneNum) {
+    u32 numBits = sizeof(u32) * 8;
+    u32 idx = sceneNum / numBits;
+    if (idx < SAVEFILE_SCENES_DISCOVERED_IDX_COUNT) {
+        u32 bit = 1 << (sceneNum - (idx * numBits));
+        return (gSaveContext.sohStats.scenesDiscovered[idx] & bit) != 0;
+    }
+    return 0;
+}
+
+void Entrance_SetSceneDiscovered(u8 sceneNum) {
+    if (Entrance_GetIsSceneDiscovered(sceneNum)) {
+        return;
+    }
+
+    u16 numBits = sizeof(u32) * 8;
+    u32 idx = sceneNum / numBits;
+    if (idx < SAVEFILE_SCENES_DISCOVERED_IDX_COUNT) {
+        u32 sceneBit = 1 << (sceneNum - (idx * numBits));
+        gSaveContext.sohStats.scenesDiscovered[idx] |= sceneBit;
+    }
+}
+
+u8 Entrance_GetIsEntranceDiscovered(u16 entranceIndex) {
+    u32 numBits = sizeof(u32) * 8;
+    u32 idx = entranceIndex / numBits;
+    if (idx < SAVEFILE_ENTRANCES_DISCOVERED_IDX_COUNT) {
+        u32 bit = 1 << (entranceIndex - (idx * numBits));
+        return (gSaveContext.sohStats.entrancesDiscovered[idx] & bit) != 0;
+    }
+    return 0;
+}
+
+void Entrance_SetEntranceDiscovered(u16 entranceIndex) {
+
+    // Skip if already set to save time from setting the connected or
+    // if this is a dynamic entrance
+    if (entranceIndex > 0x0820 || Entrance_GetIsEntranceDiscovered(entranceIndex)) {
+        return;
+    }
+
+    u16 numBits = sizeof(u32) * 8;
+    u32 idx = entranceIndex / numBits;
+    if (idx < SAVEFILE_ENTRANCES_DISCOVERED_IDX_COUNT) {
+        u32 entranceBit = 1 << (entranceIndex - (idx * numBits));
+        gSaveContext.sohStats.entrancesDiscovered[idx] |= entranceBit;
+        // Set connected
+        for (size_t i = 0; i < ENTRANCE_OVERRIDES_MAX_COUNT; i++) {
+            if (entranceIndex == gSaveContext.entranceOverrides[i].index) {
+                if (!Entrance_GetIsEntranceDiscovered(gSaveContext.entranceOverrides[i].overrideDestination)) {
+                    Entrance_SetEntranceDiscovered(gSaveContext.entranceOverrides[i].overrideDestination);
+                }
```

Comment:
i'm fine either way, but since there's already the check at the beginning of `Entrance_SetEntranceDiscovered` that early returns if `Entrance_GetIsEntranceDiscovered` is true, this could probably just be

```suggestion
                Entrance_SetEntranceDiscovered(gSaveContext.entranceOverrides[i].overrideDestination);
```

Reply:
ah yeah. Took this from 3ds rando, but that makes sense. Updated.